packages/nodes-base/nodes/Hubspot/Hubspot.node.ts

```diff
@@ -1944,6 +2233,191 @@ export class Hubspot implements INodeType {
+
+							if (additionalFields.customPropertiesUi) {
+								const customProperties = (additionalFields.customPropertiesUi as IDataObject).customPropertiesValues as IDataObject[];
+
+								if (customProperties) {
+									for (const customProperty of customProperties) {
+										properties[customProperty.property as string] = customProperty.value;
+									}
+								}
+							}
+
+							const endpoint = `/crm/v3/objects/${customObjectType}`;
+
+							responseData = await hubspotApiRequest.call(this, 'POST', endpoint, { properties });
+						}
+						if (operation === 'update') {
+							const idProperty = this.getNodeParameter('idProperty', i) as string;
+							const objectId = this.getNodeParameter('objectId', i) as string;
+							const updateFields = this.getNodeParameter('updateFields', i) as IDataObject;
+
+							const properties: IDataObject = {};
+							if (updateFields.customPropertiesUi) {
+								const customProperties = (updateFields.customPropertiesUi as IDataObject).customPropertiesValues as IDataObject[];
+
+								if (customProperties) {
+									for (const customProperty of customProperties) {
+										properties[customProperty.property as string] = customProperty.value;
+									}
+								}
+							}
+
+							const endpoint = `/crm/v3/objects/${customObjectType}/${objectId}`;
+							responseData = await hubspotApiRequest.call(this, 'PATCH', endpoint, { properties }, idProperty ? { idProperty } : {});
+						}
+						if (operation === 'upsert') {
+							const idProperty = this.getNodeParameter('idProperty', i) as string;
+							const objectId = this.getNodeParameter('objectId', i) as string;
+							const additionalFields = this.getNodeParameter('additionalFields', i) as IDataObject;
+
+							const properties: Record<string, unknown> = {};
+							if (additionalFields.customPropertiesUi) {
+								const customProperties = (additionalFields.customPropertiesUi as IDataObject).customPropertiesValues as IDataObject[];
+
+								if (customProperties) {
+									for (const customProperty of customProperties) {
+										properties[customProperty.property as string] = customProperty.value;
+									}
+								}
+							}
+							if (idProperty) {
+								properties[idProperty] = objectId;
+							}
+
+							if (!Object.keys(properties).length) {
+								throw new NodeOperationError(this.getNode(), 'You need to provide at least one property to update');
+							}
+
+							try {
+								const endpoint = `/crm/v3/objects/${customObjectType}/${objectId}`;
+								responseData = await hubspotApiRequest.call(this, 'PATCH', endpoint, { properties }, idProperty ? { idProperty } : {});
+							} catch (error) {
+								if ((error as NodeApiError).httpCode !== '404') {
+									throw error;
+								}
+								if (!idProperty) {
+									throw new NodeOperationError(this.getNode(), 'A custom id property needs to be used when creating a new object');
+								}
```

Comment:
why do custom id is needed? it brakes functionality of upsert, without this check it works as expected, creates new object when it not exist and updates if it is found

Reply:
When creating an object we cannot choose the HubSpot objectid, as it is always randomly generated. So we cannot create an object with a specific HubSpot objectid.